### PR TITLE
Add application OpenAPI definition

### PR DESCRIPTION
**Pull request by MikeRalphson**
First cut of application API OpenAPI definition.

* May need some additional `minLength` / `maxLength` properties
* I've described the properties from HAL which appear in the responses (e.g. `_embedded` and `_links`), let me know if anything else needs documenting such as 'curies' or taking descriptions from the HAL specification
* `_links` don't seem to appear in xml responses, but I've re-used the same schema for simplicity

### CODE REVIEW

**Comment by ChrisGuzman** [comment]
Looks good to me. I'm unable to get the XML responses to to render in developer.nexmo.com but I think that's a bug with the parser. Are there any specific questions I can help answer @MikeRalphson ? Feel free to comment on specific lines where you think I should pay close attention to.

File: definitions/application.yml [hunk 1]
```diff
@@ -0,0 +1,397 @@
...
+          application/x-www-form-urlencoded:
+            schema:
+              type: object
+              required:
+                - name
+                - type
+                - answer_url
+                - event_url
+              properties:
+                name:
```
Comment (MikeRalphson) [inline definitions/application.yml]: Are there any minLength/maxLength constraints for the `name` property?

File: definitions/application.yml [hunk 2]
```diff
@@ -0,0 +1,397 @@
...
+                  type: string
+                  example: POST
+      responses:
+        '201':
+          description: Created
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/application'
+            application/xml:
```
Comment (MikeRalphson) [inline definitions/application.yml]: This is possibly the reason the xml responses are not rendering. This is the media-type actually returned by the live API, but the renderer may be looking for `text/xml` ?

Comment (ChrisGuzman) [reply definitions/application.yml]: Yup the renderer was looking for text/xml. In your opinion what's the best solution? Change the OAS doc to say `text/xml` or change the parser to understand `application/xml `?

Comment (MikeRalphson) [reply definitions/application.yml]: I'd definitely fix the parser/renderer to understand both `text/xml` and `application/xml`. The OAS document should reflect what the API is really sending - that allows users to take your OAS document and use it with other tools.

**Comment by mheap** [comment]
> _links don't seem to appear in xml responses, but I've re-used the same schema for simplicity

@MikeRalphson It will mean more duplication, but I think that the OAS should specify what is returned by the production API rather than what's easy for us to maintain. Experiencing pain documenting APIs is a good indicator that we should fix the production API

**Comment by MikeRalphson** [comment]
@mheap - understood, Will amend.

**Comment by MikeRalphson** [comment]
@mheap response models split into base and xml variants where needed.

**Pull request merged**